--- linux-2.6.30/drivers/video/backlight/tdo24m.c.orig  2009-06-23
15:08:02.000000000 +0300
+++ linux-2.6.30/drivers/video/backlight/tdo24m.c       2009-06-23
15:09:05.000000000 +0300
@@ -356,7 +356,7 @@ static int __devinit tdo24m_probe(struct
        lcd->power = FB_BLANK_POWERDOWN;
        lcd->mode = MODE_VGA;   /* default to VGA */
                
-       lcd->buf = kmalloc(TDO24M_SPI_BUFF_SIZE, sizeof(GFP_KERNEL));
+       lcd->buf = kmalloc(TDO24M_SPI_BUFF_SIZE, GFP_KERNEL);
        if (lcd->buf == NULL) {
               kfree(lcd);
               return -ENOMEM;
